Packs/MicrosoftGraphTeams/Integrations/MicrosoftGraphTeams/MicrosoftGraphTeams.py

```diff
@@ -0,0 +1,736 @@
+    chat_id = args.get('chat_id')
+    user_id = args.get('user_id')
+    share_history = args.get('share_history') == 'true'
+
+    client.add_member(chat_id, user_id, share_history)
+
+    human_readable = tableToMarkdown(
+        'Member has been added successfully',
+        {
+            'Chat ID': chat_id,
+            'User ID': user_id,
+            'Share history': share_history
+        },
+        headers=['Chat ID', 'User ID', 'Share history'],
+        removeNull=True
+    )
+
+    entry_context = {}  # type: ignore
+
+    return_outputs(human_readable, entry_context)
+
+
+def list_messages_command(client: MsGraphClient, args):
+    user_id = args.get('user_id')
+    chat_id = args.get('chat_id')
+    limit = args.get('limit')
+
+    raw_response = client.list_messages(chat_id, user_id=user_id, limit=limit)
+    # return_outputs(raw_response)
+    last_page_response = raw_response[len(raw_response) - 1]
+    metadata = ''
+    next_page = last_page_response.get('@odata.nextLink')
+    if next_page:
+        metadata = '\nPay attention there are more results than shown'
+
+    message_context = build_message_object(raw_response, chat_id)
+    entry_context = {}
+    if message_context:
+        entry_context = {'MSGraphTeamsChatMessage(val.ID === obj.ID)': message_context}
+        if next_page:
+            # .NextPage.indexOf(\'http\')>=0 : will make sure the NextPage token will always be updated because it's a url
+            entry_context['MSGraphTeamsChatMessage(val.NextPage.indexOf(\'http\')>=0)'] = {'NextPage': next_page}
+
+        # human_readable builder
+        human_readable_header = f'{len(message_context)} messages found {metadata}' if metadata \
+            else f'Total of {len(message_context)} messages found'
+        human_readable = tableToMarkdown(
+            human_readable_header,
+            message_context,
+            headers=['Body', 'Created', 'From', 'ID', 'ChatID']
+        )
+    else:
+        human_readable = '### No messages were found'
+    return_outputs(human_readable, entry_context, raw_response)
+
+
+def send_message_command(client: MsGraphClient, args):
+    chat_id = args.get('chat_id')
+    body = args.get('body')
+
+    raw_response = client.send_message(chat_id, body)
+    message_context = build_message_object(raw_response, chat_id)
+    entry_context = {}
+    if message_context:
+        entry_context = {'MSGraphTeamsChatMessage(val.ID === obj.ID)': message_context}  # human_readable builder
+        human_readable_header = f'The message was send'
+        human_readable = tableToMarkdown(
+            human_readable_header,
+            message_context,
+            headers=['Body', 'Created', 'From', 'ID', 'ChatID']
+        )
+    else:
+        human_readable = '### No messages were send'
+    return_outputs(human_readable, entry_context, raw_response)
+
+
+def run_command(commands, command, client: MsGraphClient, args, tries):
```

Comment:
we have experienced some issues where if you create a chat or add a member using their userPrincipalName it would error the first time saying it was unable to find the user, while second time would be successful. We were unable to determine why, which is the reason for the addition of tries.